app/main.py

```diff
@@ -1,13 +1,28 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    # formats a single error:
+    return {
+        "line": error["line_number"],
+        "column": error["column_number"],
+        "message": error["text"],
+        "name": error["code"],
+        "source": "flake8"  # add key "source" : value = "flake8"
+    }
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    #  formats all errors
+    return {
+        "errors": [
+            format_linter_error(err) for err in errors
+        ],
+        "path": file_path,
+        "status": "passed" if errors == [] else "failed"
```

Comment:
You could use a boolean evaluation of the empty list here, consider rewriting this part without `==` assuming that the empty list is faulty.